### DISCUSSION TITLE
Redoc/Swagger Offline Support

### DISCUSSION THREAD

**Discussion by gaby** [ag2ai/faststream#1865] [General]
I haven't check if this is the case with FastStream, but by default the resources used by redoc/swagger pages require an internet connection. There's workarounds and packages that make this easier.

If it's the case, it might be worth applying those fixes to FastStream.

Related issues:
- https://github.com/fastapi/fastapi/issues/4307
- https://github.com/fastapi/fastapi/issues/4950

Potential fix via package: https://github.com/turettn/fastapi_offline

**Comment by gaby** [comment]
@Lancetnik Thoughts?

**Comment by Lancetnik** [comment]
You can already setup local source for **AsyncAPI** component - https://github.com/airtai/faststream/blob/main/faststream/asyncapi/site.py#L31-L32

But, we plan to build our own component fork to static and distribute it with the package

**Reply by gaby** [reply]
Great idea 💪
